### PR TITLE
Add PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by amandahla**
This PR adds a template for future PRs.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+### Library Changes
+
+<!-- Any changes to charm libraries -->
+
+### Checklist
+
+- [ ] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
+- [ ] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
+- [ ] The changes are compliant with [ISD014 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
+
```
Comment (gtrkiller) [inline .github/PULL_REQUEST_TEMPLATE.md:31]: I've seen other charm PR templates have more than 3 checks. Is this intended?

Comment (amandahla) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: @jdkandersson Should we use the following items in the template?

```
+ - [ ] The documentation is generated using `src-docs`
+ - [ ] The documentation for charmhub is updated.
+ - [ ] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

Comment (jdkandersson) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: yes could you please add them to the spec as well?

Comment (amandahla) [reply .github/PULL_REQUEST_TEMPLATE.md]: Changed. I just didn't change the spec because it might restart the approval process.

**Review by yanksyoon — APPROVED** [review]

**Pull request merged**